packages/@lwc/synthetic-shadow/src/faux-shadow/slot.ts

```diff
@@ -107,74 +86,115 @@ function getFilteredSlotFlattenNodes(slot: HTMLElement): Node[] {
     );
 }
 
-interface AssignedNodesOptions {
-    flatten?: boolean;
+export function assignedSlotGetterPatched(this: Element): HTMLSlotElement | null {
+    const parentNode = parentNodeGetter.call(this);
+    /**
+     * if it doesn't have a parent node,
+     * or the parent is not an slot element
+     * or they both belong to the same template (default content)
+     * we should assume that it is not slotted
+     */
+    if (
+        isNull(parentNode) ||
+        !isSlotElement(parentNode) ||
+        getNodeNearestOwnerKey(parentNode) === getNodeNearestOwnerKey(this)
+    ) {
+        return null;
+    }
+    return parentNode as HTMLSlotElement;
 }
 
-export function PatchedSlotElement(elm: HTMLSlotElement): HTMLSlotElementConstructor {
-    const Ctor = PatchedElement(elm) as HTMLSlotElementConstructor;
-    const { addEventListener: superAddEventListener } = elm;
-    return class PatchedHTMLSlotElement extends Ctor {
-        addEventListener(
+defineProperties((window as any).HTMLSlotElement.prototype, {
+    addEventListener: {
+        value(
             this: HTMLSlotElement,
             type: string,
             listener: EventListener,
             options?: boolean | AddEventListenerOptions
         ) {
-            if (type === 'slotchange' && !getInternalField(this, SlotChangeKey)) {
-                if (process.env.NODE_ENV === 'test') {
-                    /* eslint-disable-next-line no-console */
-                    console.warn(
-                        'The "slotchange" event is not supported in our jest test environment.'
-                    );
-                }
+            // super.addEventListener - but that doesn't work with typescript
+            HTMLElement.prototype.addEventListener.call(this, type, listener, options);
```

Comment:
Why not import addEventListener from ../env/element?

Reply:
it is a little bit more tricky, maybe we can accommodate that in another refactor?